src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -55,72 +111,656 @@ static InitChainEntry D_80936754[] = {
+        this->actor.home.rot.y = this->actor.shape.rot.y;
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+    }
+    if (EnKusa_SnapToFloor(this, globalCtx, 0.0f) == 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    if (EnKusa_GetWaterBox(this, globalCtx)) {
+        this->isInWater |= 1;
+    }
+
+    this->objIndex = Object_GetIndex(&globalCtx->objectCtx, objectIds[(GET_KUSA_TYPE(&this->actor))]);
+    if (this->objIndex < 0) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    EnKusa_SetupWaitObject(this);
+    if (D_809366B4) {
+        D_80936CD8 = ((u32)Rand_Next() >> 0x10);
+        D_80936CDA = ((u32)Rand_Next() >> 0x10);
+        D_80936CDC = ((u32)Rand_Next() >> 0x10);
+        D_80936CDE = ((u32)Rand_Next() >> 0x10);
+        D_80936CE0 = ((u32)Rand_Next() >> 0x10);
+        D_809366B4 = false;
+        EnKusa_Sway();
+        kusaGameplayFrames = globalCtx->gameplayFrames;
+    }
+    this->kusaMtxIdx = D_809366B0 & 7;
+    D_809366B0++;
+}
+
+void EnKusa_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    EnKusa* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnKusa_SetupWaitObject(EnKusa* this) {
+    this->actionFunc = EnKusa_WaitObject;
+}
+
+/**
+ * @brief Waits for object to be loaded then draws the appropriate model
+ *
+ */
+void EnKusa_WaitObject(EnKusa* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objIndex)) {
+        s32 kusaType = GET_KUSA_TYPE(&this->actor);
+
+        if (this->isCut) {
+            EnKusa_SetupCut(this);
+        } else {
+            EnKusa_SetupInteract(this);
+        }
+        if (kusaType == ENKUSA_TYPE_BUSH) {
+            this->actor.draw = EnKusa_DrawBush;
+        } else {
+            this->actor.draw = EnKusa_DrawGrass;
+        }
+        this->actor.objBankIndex = this->objIndex;
+        this->actor.flags &= ~ACTOR_FLAG_10;
+    }
+}
+
+void EnKusa_SetupInteract(EnKusa* this) {
+    this->actionFunc = EnKusa_WaitForInteract;
+    this->actor.flags &= ~ACTOR_FLAG_10;
+}
+
+/**
+ * @brief Handles the main logic and control flow for all kusa types
+ *
+ */
+void EnKusa_WaitForInteract(EnKusa* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        EnKusa_SetupLiftedUp(this);
+        SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 20, NA_SE_PL_PULL_UP_PLANT);
+        this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
+
+    } else if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        EnKusa_SpawnFragments(this, globalCtx);
+        EnKusa_DropCollectible(this, globalCtx);
+        SoundSource_PlaySfxAtFixedWorldPos(globalCtx, &this->actor.world.pos, 20, NA_SE_EV_PLANT_BROKEN);
+
+        if ((KUSA_GET_PARAMS_04(&this->actor))) {
```

Comment:
Well, surely you can name this one more descriptively? `KUSA_GET_SHOULD_SPAWN_BUGS`? It's used once so it's okay for it to be a bit longer.